subworkflows/local/bam_variant_calling_single_strelka/main.nf

```diff
@@ -4,84 +4,55 @@ include { STRELKA_GERMLINE as STRELKA_SINGLE      } from '../../../modules/nf-co
-    STRELKA_SINGLE.out.vcf.branch{
-            intervals:    it[0].num_intervals > 1
-            no_intervals: it[0].num_intervals <= 1
-        }.set{strelka_vcf}
-
-    STRELKA_SINGLE.out.genome_vcf.branch{
-            intervals:    it[0].num_intervals > 1
-            no_intervals: it[0].num_intervals <= 1
-        }.set{strelka_genome_vcf}
-
-    MERGE_STRELKA(
-        strelka_vcf.intervals
-            .map{ meta, vcf ->
-                new_meta = [
-                                id:             meta.sample,
-                                num_intervals:  meta.num_intervals,
-                                patient:        meta.patient,
-                                sample:         meta.sample,
-                                sex:            meta.sex,
-                                status:         meta.status
-                            ]
-
-                [groupKey(new_meta, meta.num_intervals), vcf]
-            }.groupTuple(),
-        dict.map{ it -> [[id:it[0].baseName], it]})
-
-    MERGE_STRELKA_GENOME(
-        strelka_genome_vcf.intervals
-            .map{ meta, vcf ->
-
-                [groupKey([
-                            id:             meta.sample,
-                            num_intervals:  meta.num_intervals,
-                            patient:        meta.patient,
-                            sample:         meta.sample,
-                            sex:            meta.sex,
-                            status:         meta.status,
-                        ],
-                        meta.num_intervals),
-                vcf]
-
-            }.groupTuple(),
-        dict.map{ it -> [[id:it[0].baseName], it]})
-
-    // Mix output channels for "no intervals" and "with intervals" results
-    // Only strelka variant vcf should get annotated
-    strelka_vcf = Channel.empty().mix(
-                    MERGE_STRELKA.out.vcf,
-                    strelka_vcf.no_intervals)
-                .map{ meta, vcf ->
-                    [[
-                        id:             meta.sample,
-                        num_intervals:  meta.num_intervals,
-                        patient:        meta.patient,
-                        sample:         meta.sample,
-                        sex:            meta.sex,
-                        status:         meta.status,
-                        variantcaller:  "strelka"
-                    ], vcf]
-                }
-
-    ch_versions = ch_versions.mix(MERGE_STRELKA.out.versions)
-    ch_versions = ch_versions.mix(MERGE_STRELKA_GENOME.out.versions)
-    ch_versions = ch_versions.mix(STRELKA_SINGLE.out.versions)
+    versions = Channel.empty()
+
+    // Combine cram and intervals for spread and gather strategy
+    cram_intervals = cram.combine(intervals)
+        // Move num_intervals to meta map
+        .map{ meta, cram, crai, intervals, intervals_index, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals, intervals_index ] }
```

Comment:
Won't this meta handling cause problems because of the lack of cloning?

Reply:
It shouldn't, but I'll check with @robsyme

Reply:
I think the new magic addition doesn't. Mahesh introduced it on slack

Reply:
No cloning necessary.
The plus operator returns a _new_ map object and does not modify the original. Documentation [here](https://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/Map.html#plus(java.util.Map)).

Reply:
Oooh thanks! I'm learning here